src/VisualStudio/Core/Impl/ProjectSystem/CPS/TempPECompiler.cs

```diff
@@ -49,16 +49,31 @@ public async Task<bool> CompileAsync(IWorkspaceProjectContext context, string ou
                 cancellationToken.ThrowIfCancellationRequested();
             }
 
-            var options = project.LanguageServices.GetRequiredService<ICompilationFactoryService>().GetDefaultCompilationOptions()
+            // We need to inherit most of the projects options, mainly for VB (RootNamespace, GlobalImports etc.), but we need to override about some specific things surrounding the output
+            var options = project.CompilationOptions
                     // copied from the old TempPE compiler used by legacy, for parity.
                     // See: https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/VisualStudio/CSharp/Impl/ProjectSystemShim/TempPECompilerService.cs#L58
                     .WithAssemblyIdentityComparer(DesktopAssemblyIdentityComparer.Default)
                     .WithSourceReferenceResolver(SourceFileResolver.Default)
                     .WithXmlReferenceResolver(XmlFileResolver.Default)
-                    // we always want to produce a DLL
-                    .WithOutputKind(OutputKind.DynamicallyLinkedLibrary);
+                    // We always want to produce a debug, AnyCPU DLL
+                    .WithOutputKind(OutputKind.DynamicallyLinkedLibrary)
+                    .WithPlatform(Platform.AnyCpu)
+                    .WithOptimizationLevel(OptimizationLevel.Debug)
+                    // Turn off any warnings as errors just in case
+                    .WithGeneralDiagnosticOption(ReportDiagnostic.Suppress)
+                    .WithReportSuppressedDiagnostics(false)
+                    .WithSpecificDiagnosticOptions(null)
+                    // Turn off any signing and strong naming
+                    .WithDelaySign(false)
+                    .WithCryptoKeyFile(null)
+                    .WithPublicSign(false)
+                    .WithStrongNameProvider(null);
```

Comment:
You shouldn't have to remove the provider.

Reply:
Is it a problem that I am? I just went through and tried to un-set all of the things that could possibly be set. I admit to not having a deep understanding of the overlaps here.

Reply:
I wasn't sure if a null provider might also change other subtle things. @tmat would know.